cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesAdder.java

```diff
@@ -100,20 +221,103 @@ public static void add(Network n, CgmesModel cgmes) {
                 tapSteps.add(p);
             }
         }
-        cgmes.add(CgmesSubset.STATE_VARIABLES, "SvTapStep", tapSteps);
+
+        for (ThreeWindingsTransformer t : network.getThreeWindingsTransformers()) {
+            PropertyBag p = new PropertyBag(svTapStepProperties);
+            Arrays.asList(t.getLeg1(), t.getLeg2(), t.getLeg3()).forEach(leg -> {
+                if (leg.getPhaseTapChanger() != null) {
+                    p.put(tapChangerPositionName, is(leg.getPhaseTapChanger().getTapPosition()));
+                    p.put(CgmesNames.TAP_CHANGER, cgmes.phaseTapChangerForPowerTransformer(t.getId()));
+                    tapSteps.add(p);
+                } else if (leg.getRatioTapChanger() != null) {
+                    p.put(tapChangerPositionName, is(leg.getRatioTapChanger().getTapPosition()));
+                    p.put(CgmesNames.TAP_CHANGER, cgmes.ratioTapChangerForPowerTransformer(t.getId()));
+                    tapSteps.add(p);
+                }
+            });
+        }
+
+        cgmes.add(originalSVcontext, "SvTapStep", tapSteps);
+    }
+
+    private void addStatusToCgmes() {
+        // create SvStatus, iterate on Connectables, check Terminal status, add
+        // to SvStatus
+        PropertyBags svStatus = new PropertyBags();
+        Set<String> addedConnectables = new HashSet<>();
+        for (VoltageLevel v : network.getVoltageLevels()) {
+            for (Connectable<?> c : v.getConnectables()) {
+                // need to check if connectable was already added
+                if (!addedConnectables.contains(c.getId())) {
```

Comment:
@mathbagu We discussed it but I don't really like that `getConnectables()` does not ensure that the elements are distinct
@kaltakovae Do you have a case in which the same `Connectable` is met several times? In any case, I think you should drop this check, I will make another PR to ensure the list/stream is distinct. Besides, you can use the stream API here.

Reply:
Currently we observe multiple duplicates, e.g. for each Line or Transformer getConnectables() duplicates results. We've tested with MicroGrid and HOPS test cases.
But, even if getConnectables() results in distincts, we still need the check for duplicates here:
Without this check any Connectables which are in different VoltageLevels (e.g Transformers) will be duplicated or triplicated.

Reply:
Thank you for your input! I made a PR to check for duplicates in a same voltage level. As for your code, thank your for the clarification! I think you can optimize it as such:
```java
PropertyBags svStatus = new PropertyBags();
        network.getVoltageLevelStream()
                .flatMap(VoltageLevel::getConnectableStream)
                .distinct()
                .forEach(c -> {
                    PropertyBag p = new PropertyBag(SV_SVSTATUS_PROPERTIES);
                    for (Terminal t : ((Connectable<?>) c).getTerminals()) {
                        p.put(IN_SERVICE, Boolean.toString(t.isConnected()));
                    }
                    p.put(CgmesNames.CONDUCTING_EQUIPMENT, c.getId());
                    svStatus.add(p);
                });
```

Reply:
About finding duplicates:
There is no `getConnectables` at `Network` level, so to obtain all connectables of the network we must iterate first over the voltage levels, then its connectables.
This means that we have to account for potential duplicates: lines and 2w transformers will be found twice, once for every end, and 3w transformers will appear three times during the iteration.
So, unless we have a `getConnectables` at `Network` level, we have to take into account for potential duplicates. 
An alternative could be to iterate over specific types of equipment (`getLines`, `getTransformers`, `getGenerators`, `getLoads`, ...) but that would mean to have a long list of method calls, and leave the risk of omitting new types when they are added to the Network (in the future). I mean, leaving a long list of specific dependencies when a generic treatment over all kind of equipment can be done.

Reply:
@MioRtia  Thank you Miora, this mathes perfectly to what we wanted. I've done the change.